lib/http/node.js

```diff
@@ -100,6 +100,13 @@ function isBypassProxy(proxy, destination) {
  * @inheritDoc
  */
 NodeHttpClient.prototype.getAgent = function (parsedUrl, proxy, mock) {
+  if (!proxy && GlobalConfig.isEnvProxyActive()) {
+    const isHttps = parsedUrl.protocol === 'https:';
+    proxy = Util.getProxyEnv(isHttps);
```

Comment:
@sfc-gh-dszmolka, that is a good question. Actually, the original request for this ticket is that we can use HTTP_PROXY and HTTPS_PROXY env variables like Axios. However, we have one problem. Axios proxy keeps changing the proxy info based on the location of URL (If the location is http, use HTTP_PROXY. If https, use HTTS_PROXY) call. However, our driver only supports one proxy that a user configured in the connectionConfig.

Currently, I am not sure which workflow we should implement. Let's wait for @sfc-gh-pmotacki response.

Thank you for your questions.

Reply:
Hello @sfc-gh-dszmolka, please take a look at https://github.com/snowflakedb/snowflake-connector-nodejs/pull/938/files#diff-f7fc5e0118097ba33301dd904921ab1c12e77854a9493c441aaf79aadb00be69R1398. This is how Axios works. However, we can change the workflow if we want to use the schema 'http' as default in the driver. Let's wait for @sfc-gh-pmotacki response.

Reply:
FYI:
This is the code from proxy-from-env that Axios uses: https://github.com/Rob--W/proxy-from-env/blob/master/index.js#L44

Axios: https://github.com/axios/axios/blob/v1.x/lib/adapters/http.js#L86

Reply:
thanks for the answers @sfc-gh-ext-simba-jy ! for the 'this is how axios does it' topics - this very PR is necessary because the stuff axios does with proxies, and it doesn't work, and causes problems :)